python/lsst/obs/base/makeRawVisitInfoViaObsInfo.py

```diff
@@ -70,7 +70,11 @@ def __init__(self, log=None):
         self.log = log
 
     def __call__(self, md, exposureId=None):
-        """Construct a VisitInfo and strip associated data from the metadata.
+        """Construct a `~lsst.afw.image.VisitInfo` from a header.
+
+        Does not strip the associated metadata keywords from the header
+        since a VisitInfo does not include all information stored in
+        `~astro_metadata_translator.ObservationInfo`.
```

Comment:
I'm saying that if you create an ObservationInfo, then strip all the headers out that were used to create it, then there are no headers left that would allow you to translate the header again. It seems that obs_lsst is running makeRawVisitInfo multiple times on the same header and the second time through it doesn't work because all the useful headers are missing. Regardless, stripping here is a bad idea because what you get back is a subset of ObservationInfo. If VisitInfo was turned into a C++ AFW representation of an ObservationInfo then removing the headers would be more justifiable (but would still fail if you did what obs_lsst does).